libfreerdp/crypto/tls.c

```diff
@@ -737,10 +737,14 @@ static int tls_do_handshake(rdpTls* tls, BOOL clientMode)
 
 		do
 		{
-			status = poll(&pollfds, 1, 10 * 1000);
+			status = poll(&pollfds, 1, 10);
 		}
 		while ((status < 0) && (errno == EINTR));
 
+		if (status == 0) {
+			status = WAIT_TIMEOUT;
```

Comment:
WAIT_TIMEOUT is used below only in case of "_WIN32" and I believe "poll" is not going to be used in that case, so there seem to be no need for such fixup.

Reply:
I am not so sure about whether HAVE_POLL_H is always undefined in Win32.
IMO, this statement at worst has no effect, and at best avoids an error in Windows. @akallabeth wat do you think?

Reply:
@ccpp This statement creates unnecessary confusion as if "poll" is supported on _WIN32 and it's not.
In case of _WIN32 the code on line 724 does "BIO_get_event(tls->bio, &event);". And "poll" uses "fd", which is left uninitialized. If poll returns "1", the _WIN32 specific check on line 767 will fail as well.
I think, these series of ifndefs need to be rearranged to more clearly indicate supported configurations, but in mean time we shouldn't make it even more confusing.

Reply:
@p-pautov fair point!